fixes: python -m build fails

arenevier opened the pull request:
python -m build fails with:
> running check
> error: The docutils package is needed.

since check.restructuredtext is set to 1 in setup.cfg, setuptools with require docutils. I think this is needed to parse the structured text README.
That dependency needs to be specify in pyproject.toml file. It is recommended practice to have a pyproject.toml file anyway. https://packaging.python.org/en/latest/guides/modernize-setup-py-project/#should-pyproject-toml-be-added

arenevier committed:
fixes: python -m build fails

python -m build fails with:
> running check
> error: The docutils package is needed.

since check.restructuredtext is set to 1 in setup.cfg, setuptools with
require docutils. I think this is needed to parse the structured text
README.
That dependency needs to be specify in pyproject.toml file. It is recommended practice to have a pyproject.toml file anyway.
https://packaging.python.org/en/latest/guides/modernize-setup-py-project/#should-pyproject-toml-be-added

ggrossetie commented:
@arenevier does it make sense to add a CI with this command `python -m build`? I want to make sure that we don't break it.

arenevier committed:
new tox environments: sdist and wheel

Build a source distribution and a binary distribution when running tox.
This makes sure we don't break those

arenevier commented:
@ggrossetie : good idea. Instead of adding a new CI step,  I added two new targets to tox. One to build the source distribution (the one currently failing). And one to build binary distribution (which is not currently broken).

ggrossetie commented:
Just to be sure, the two new targets do not run on the CI? If not, I think we should run them to make sure that a future pull request won't break them.

ggrossetie commented:
Also, I just noticed that we don't have a tox environment for Pillow 10.x

arenevier committed:
run the new tox targets in the CI

arenevier commented:
> Also, I just noticed that we don't have a tox environment for Pillow 10.x

I can make a new PR to add it. And maybe python 3.12. I will wait for this one to be merged first, otherwise there will be merge conflicts.

arenevier commented:
> Just to be sure, the two new targets do not run on the CI? If not, I think we should run them to make sure that a future pull request won't break them.

Fixed. They run now with the latest commit of the branch.

ggrossetie commented:
Awesome, thanks 🎉
